### ISSUE TITLE
about "NaNs in the encoded observations"

### ISSUE THREAD

**Issue by ccconquer** [sisl/SceneInformer#6]
Thanks for your contribution！
How can I fix this problem, can you give me some advice?
![image](https://github.com/user-attachments/assets/46de487a-c6de-4ebb-a1a9-5d0b61b46676)

**Comment by ccconquer** [comment]
By the way, will you upload the validation part, or how can i validate?

**Comment by BenQLange** [comment]
First, track down where the NaNs originate. Either the input observations contain some NaNs, or the weights became NaN during training due to instability or NaNs in the loss function.

I’ve used NaNs in the dataset to flag any invalid/illegal observations (e.g., far-away trajectories, occlusions) so that there is no leakage. After the mask is generated:

https://github.com/sisl/SceneInformer/blob/efce1976e939b08eb4608f4eb679a1179926e4e5/sceneinformer/model/encoder.py#L57  

They should all be set to 0 to prevent any PyTorch errors done here: https://github.com/sisl/SceneInformer/blob/efce1976e939b08eb4608f4eb679a1179926e4e5/sceneinformer/model/encoder.py#L72

The same logic applies to polylines, so this issue shouldn't occur unless something has been commented out.

A similar approach is used for the loss function (sceneinformer/model/loss.py):

https://github.com/sisl/SceneInformer/blob/efce1976e939b08eb4608f4eb679a1179926e4e5/sceneinformer/model/loss.py#L23

Check that as well.

If none of the above applies, then it is most likely caused by training instability, in which case typical solutions should be applied, such as lowering the learning rate, clipping gradients, or increasing precision.

I’ve included the hyperparameters in the config file for easy adjustment:
https://github.com/sisl/SceneInformer/blob/efce1976e939b08eb4608f4eb679a1179926e4e5/configs/scene_informer.yaml#L113-L122

Hopefully, that's helpful.

**Comment by BenQLange** [comment]
> By the way, will you upload the validation part, or how can i validate?

I'll try to find the exact script I've used. 

Validation script just compares the FDE/ADE of the generated trajectories and compares the anchor occupancy (like in the loss function).
